src/quo2/components/dividers/strength_divider/view.cljs

```diff
@@ -0,0 +1,86 @@
+(ns quo2.components.dividers.strength-divider.view
+  (:require [quo2.components.dividers.strength-divider.style :as style]
+            [quo2.components.icon :as icon]
+            [quo2.components.markdown.text :as text]
+            [quo2.foundations.colors :as colors]
+            [react-native.core :as rn]
+            [react-native.linear-gradient :as linear-gradient]
+            [react-native.svg :as svg]
+            [utils.i18n :as i18n]))
+
+(def statuses
+  {:very-weak   {:default-text (i18n/label :t/strength-divider-status-very-weak)
+                 :color        colors/danger-60
+                 :percentage   20}
+   :weak        {:default-text (i18n/label :t/strength-divider-status-weak)
+                 :color        (colors/custom-color :orange 60)
+                 :percentage   40}
+   :okay        {:default-text (i18n/label :t/strength-divider-status-okay)
+                 :color        (colors/custom-color :yellow 60)
+                 :percentage   60}
+   :strong      {:default-text (i18n/label :t/strength-divider-status-strong)
+                 :color        colors/success-60
+                 :percentage   80}
+   :very-strong {:default-text (i18n/label :t/strength-divider-status-very-strong)
+                 :color        colors/success-60
+                 :percentage   100}
+   :info        {:color colors/white-opa-40}
+   :alert       {:color colors/danger-60}})
+
+
+(defn circular-progress
+  [{:keys [color percentage]}]
+  (let [status-indicator-radius 6.5
+        status-circumference    (* 2 Math/PI status-indicator-radius)]
+    [svg/svg
+     {:view-box  "0 0 16 16"
+      :width     14.2
+      :transform [{:rotate "270deg"}]
+      :height    14.2}
+     [svg/circle
+      {:cx           8
+       :cy           8
+       :r            status-indicator-radius
+       :fill         :transparent
+       :stroke-width 1.2
+       :stroke       (colors/alpha color 0.2)}]
+     [svg/circle
+      {:cx                8
+       :cy                8
+       :r                 status-indicator-radius
+       :fill              :transparent
+       :stroke-width      1.2
+       :stroke-dasharray  status-circumference
+       :stroke-dashoffset (* (- 100 percentage) 0.01 status-circumference)
+       :stroke            color}]
+     [svg/circle]]))
+
+(defn status-indicator
+  [status]
+  (let [{:keys [color percentage]} (statuses status)]
+    (case status
+      :info  nil
+      :alert [icon/icon :i/alert
+              {:color color
+               :size  16}]
+      [circular-progress {:color color :percentage percentage}])))
+
+(defn view
+  "Options
+   - `status` Status keyword (`:very-weak`/`:weak`/`:okay`/`:strong`/`:very-strong`/`:info`/`:alert`)
+
+   `text` message string(only works when status is `info`/`alert`)"
+  [{:keys [status] :or {status :very-weak}} text]
```

Comment:
Or maybe we can use type instead of status. (Ex. Strength type weak, strong etc)